backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewEdgeService.java

```diff
@@ -0,0 +1,124 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.syson.diagram.common.view.services;
+
+import java.util.List;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.resource.ResourceSet;
+import org.eclipse.syson.services.UtilService;
+import org.eclipse.syson.sysml.AllocationUsage;
+import org.eclipse.syson.sysml.Element;
+import org.eclipse.syson.sysml.EndFeatureMembership;
+import org.eclipse.syson.sysml.Feature;
+import org.eclipse.syson.sysml.ReferenceSubsetting;
+import org.eclipse.syson.sysml.SysmlPackage;
+import org.eclipse.syson.util.SysMLMetamodelHelper;
+
+/**
+ * Edge-related Java services used by several diagrams.
+ *
+ * @author Jerome Gout
+ */
+public class ViewEdgeService {
+
+    private final UtilService utilService;
+
+    public ViewEdgeService() {
+        this.utilService = new UtilService();
+    }
+
+    /**
+     * Get all reachable allocate edge in the {@link ResourceSet} of given {@link EObject}.
+     *
+     * @param eObject
+     *            the {@link EObject} stored in a {@link ResourceSet}
+     * @return a list of allocate edge objects
+     */
+    public List<AllocationUsage> getAllReachableAllocateEdges(EObject eObject) {
+        String type = SysMLMetamodelHelper.buildQualifiedName(SysmlPackage.eINSTANCE.getAllocationUsage());
+        var allAllocationUsages = this.utilService.getAllReachable(eObject, type);
+        return allAllocationUsages.stream()
+                .filter(AllocationUsage.class::isInstance)
+                .map(AllocationUsage.class::cast)
+                .filter(au -> this.isAnAllocateEdge(au)).toList();
```

Comment:
Please put the `.toList()` on a new line.
(also true for several other occurrences in this file and other files)

Reply:
Done